futures-util/src/stream/recover.rs

```diff
@@ -0,0 +1,35 @@
+use std::marker::PhantomData;
+
+use futures_core::{Stream, Poll, Async};
+use futures_core::task;
+
+/// Future for the `recover` combinator, handling errors by converting them into
+/// an `Option<Item>`, compatible with any error type of the caller's choosing.
```

Comment:
Can you add more documentation around the `None` behavior? Specifically, that `None` causes the stream to end, rather than just skipping to the next element.